internal/pkg/workspace/workspace.go

```diff
@@ -409,15 +409,46 @@ func (ws *Workspace) DeleteWorkspaceFile() error {
 	return ws.fs.Remove(filepath.Join(CopilotDirName, SummaryFileName))
 }
 
-// ReadAddonsDir returns a list of file names under a service's "addons/" directory.
-func (ws *Workspace) ReadAddonsDir(svcName string) ([]string, error) {
+// EnvAddonsPath returns the addons/ directory file path for environments.
+func (ws *Workspace) EnvAddonsPath() (string, error) {
 	copilotPath, err := ws.copilotDirPath()
```

Comment:
I wonder if we could have gotten-rid of so many of our errors if we had a different approach to `New` in our package. For example:
```go
package workspace

// Workspace is a created copilot/ directory in a git repository.
type Workspace struct{...}

// Use returns an existing workspace and replaces the old "New" function. 
// It searches a copilot/ directory from the current wd, up to 5 levels above.
func Use() (*Workspace, error) { ... }

// Create is no longer a method and instead is a function that creates a new *Workspace in the current working directory for appName.
func Create(appName string) (*Workspace, error)
```

With the above API, I think we can completely ditch the `copilotDirPath` error because `*Workspace` will have the value stored in a field.

Let me know what you think of the above, if it sounds good we can do it as a follow-up to this PR?

Reply:
Makes sense to me!

We can use `Use()` to do what've been done by `copilotDirPath` so that the rest of the operations can proceed with the assumption that we are in a worksapce.